packages/next/server/config-shared.ts

```diff
@@ -76,6 +76,7 @@ export interface NextJsWebpackConfig {
 }
 
 export interface ExperimentalConfig {
+  newLinkBehavior?: boolean
```

Comment:
What do you think about calling this `nextLinkAutoAnchor` to make it more explicit what this is enabling?